packages/mocks/src/console-handlers/providers.ts

```diff
@@ -0,0 +1,74 @@
+import { rest, RestHandler } from 'msw';
+
+import { createResource, ForkliftResourceKind } from '@kubev2v/legacy/client/helpers';
+import { IKubeList } from '@kubev2v/legacy/client/types';
+import { CLUSTER_API_VERSION } from '@kubev2v/legacy/common/constants';
+import { IProviderObject } from '@kubev2v/legacy/queries/types';
+
+import { MOCK_INVENTORY_PROVIDERS } from '../mocks/providers.mock';
+
+import { fancyCaptureUrl } from './util';
+
+const mockClusterProviders = (namespace?: string): IKubeList<IProviderObject> => {
+  const items = Object.values(MOCK_INVENTORY_PROVIDERS)
+    .flat()
+    .filter((provider) => (namespace ? provider.namespace === namespace : true))
+    .map((provider) => provider.object);
+
+  return {
+    apiVersion: CLUSTER_API_VERSION,
+    kind: 'Providers',
+    metadata: {
+      continue: '',
+      resourceVersion: '',
+      selfLink: '/foo/list/selfLink',
+    },
+    items,
+  };
+};
+
+// these ultimately rely on @migtools/lib-ui for path generation
+const providerClusterResource = createResource(ForkliftResourceKind.Provider, '');
+
+const providerNamespacedResource = createResource(ForkliftResourceKind.Provider, ':namespace');
+
+const getInventoryApiUrl = (relativePath?: string): string =>
+  `/api/proxy/plugin/forklift-console-plugin/forklift-inventory/${relativePath || ''}`;
+
+/**
+ * These handler paths should be from the perspective of the plugin running on console.
+ *
+ * The intention is to have handlers for every providers based query and mutation
+ * present in `packages/legacy/src/queries/providers.ts`.
+ */
+export const handlers: RestHandler[] = [
```

Comment:
handlers should be on the plugin side using the method that fetches data using path and method

Reply:
I'm leaving the handlers on the mocks side since it makes sense.  There is enough code per endpoint that I'd rather keep the handlers close to the mock data and just have the extension apply them,